fsgrid.hpp

```diff
@@ -965,27 +978,60 @@ template <typename T, int stencil> class FsGrid : public FsGridTools{
       //! Copy the entire data from another FsGrid of the same signature over.
       FsGrid<T, stencil>& operator=(const FsGrid<T, stencil>& other) {
 
-         // Don't copy if sizes mismatch.
-         // (Should this instead crash the program?)
-         if(other.localSize[0] != localSize[0]   ||
-            other.localSize[1] != localSize[1]   ||
-            other.localSize[2] != localSize[2]) {
-            return *this;
-         }
+         // Fail if sizes mismatch.
+         assert(other.localSize[0] == this->localSize[0]);
+         assert(other.localSize[1] == this->localSize[1]);
+         assert(other.localSize[2] == this->localSize[2]);
          data = other.data;
+         return *this;
+      }
 
+      FsGrid<T, stencil>& operator+=(const FsGrid<T, stencil>& rhs) {
+         assert(this->data.size()==rhs.data.size());
+         for (size_t i=0; i< this->data.size(); i++){
+            this->data[i] = this->data[i]+rhs.data[i];
+         }
+         return *this;
+      }
+
+      FsGrid<T, stencil>& operator-=(const FsGrid<T, stencil>& rhs) {
+         assert(this->data.size()==rhs.data.size());
+         for (size_t i=0; i< this->data.size(); i++){
+            this->data[i] = this->data[i]-rhs.data[i];
+         }
+         return *this;
+      }
+
+      FsGrid<T, stencil>& operator*=(const FsGrid<T, stencil>& rhs) {
+         assert(this->data.size()==rhs.data.size());
+         for (size_t i=0; i< this->data.size(); i++){
+            this->data[i] = this->data[i]*rhs.data[i];
+         }
+         return *this;
+      }
+
+      FsGrid<T, stencil>& operator/=(const FsGrid<T, stencil>& rhs) {
+         assert(this->data.size()==rhs.data.size());
+         for (size_t i=0; i< this->data.size(); i++){
+            this->data[i] = this->data[i]/rhs.data[i];
+         }
+         return *this;
+      }
+
+      template <typename S, typename std::enable_if<std::is_arithmetic<S>::value>::type * = nullptr>
```

Comment:
Nice. Your C++ kung-fu is starting to be scary!

Reply:
That's good right?